application/forms/ProcessForm.php

```diff
@@ -22,31 +22,27 @@ class ProcessForm extends QuickForm
     /** @var BpNode */
     protected $node;
 
-    protected $objectList = array();
-
-    protected $processList = array();
-
     /** @var SessionNamespace */
     protected $session;
 
     public function setup()
     {
-        if ($this->node === null) {
-            $this->addElement('text', 'name', array(
-                'label'        => $this->translate('ID'),
-                'required'     => true,
-                'description' => $this->translate(
-                    'This is the unique identifier of this process'
-                ),
-            ));
-        } else {
+        if ($this->node !== null) {
             $this->addHtml(
                 '<h2>' . $this->getView()->escape(
                     sprintf($this->translate('Modify "%s"'), $this->node->getAlias())
                 ) . '</h2>'
             );
         }
 
+        $this->addElement('text', 'name', [
+            'label'         => $this->translate('ID'),
+            'value'         => (string) $this->node,
+            'required'      => $this->node !== null,
+            'readonly'      => $this->node === null,
```

Comment:
Setting `readonly` to anything else than `null` causes it to be effective. Ensure that this is only effective when you want. (Ignoring the fact, that this form isn't used for *new* processes for now)